test/integration/builder/selects.js

```diff
@@ -540,6 +540,85 @@ module.exports = function(knex) {
           });
       });
 
+      it('allows parameter to be a query builder', function() {
+        var whereClause = knex.where({'id': 0});
+        return knex('accounts')
+          .where(whereClause)
+          .select()
+          .testSql(function(tester) {
+            tester(
+              'mysql',
+              'select * from `accounts` where `id` = ?',
+              [0],
+              []
+            );
+            tester(
+              'postgresql',
+              'select * from "accounts" where "id" = ?',
+              [0],
+              []
+            );
+            tester(
+              'sqlite3',
+              'select * from "accounts" where "id" = ?',
+              [0],
+              []
+            );
+            tester(
+              'oracle',
+              'select * from "accounts" where "id" = ?',
+              [0],
+              []
+            );
+            tester(
+              'mssql',
+              'select * from [accounts] where [id] = ?',
+              [0],
+              []
+            );
+          });
+      });
+    });
+
+    it('can have a where clause with complex query builder', function() {
+      var simpleWhereClause = knex.where('id', 4);
+
+      var complexWhereClause = knex
+        .where('logins', '>', 1)
+        .whereIn('id', function () {
+          this.select('id').from('accounts').whereNotNull('logins').where(simpleWhereClause)
+          .where(function () {
+            this.where(simpleWhereClause);
+          });
+        });
+
+      knex('accounts').where(simpleWhereClause).where(complexWhereClause).orWhere(simpleWhereClause)
```

Comment:
Could you please check the generated query from .toSQL() output... like wth tester function does? The query executed here is so insane that parts of it might not be correct but it still would create the same results.